### PR TITLE
Add category management screens

### PR DESCRIPTION
**Pull request by dNitza**
This PR lets us create, edit and delete categories.

If a category is assigned to a post, and then deleted, it will be removed from the post (its record in the join column will be destroyed).

screen shot 2016-05-17 at 4 06 31 pm

### CODE REVIEW

File: apps/admin/web/routes/categories.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        r.view "categories.new"
+      end
+
+      r.on ":slug" do |slug|
+        r.get "edit" do
+          r.view "categories.edit", slug: slug
+        end
+
+        r.post do
+          if r["_method"] == "delete"
```
Comment (dNitza) [inline apps/admin/web/routes/categories.rb]: I'm not too happy with this. I had tried using the `all_verbs` plugin for Roda, but it as far as I can tell, it expects `request_method` to be set in the environment and not as part of a post variable. So, perhaps this is an acceptable work around? Or we could use a variation of the `data-view-confirm-action` method in Roneo to send the delete request.

Comment (timriley) [inline apps/admin/web/routes/categories.rb]: Are you also using `use Rack::MethodOverride` inside `Admin::Application`, @dNitza? I think that's what handles the `_method` post params and adjusts the request into a method that properly matches.

File: apps/admin/lib/admin/categories/forms/edit_form.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        prefix :category
+
+        define do
+          text_field :name, label: "Name"
+          text_field :slug, label: "Slug"
+        end
+      end
+    end
+  end
+end
```
Comment (timriley) [inline apps/admin/lib/admin/categories/forms/edit_form.rb]: I guess we could merge the create and edit forms if we expect they'll remain identical?

File: apps/admin/lib/admin/categories/operations/delete.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        include Dry::ResultMatcher.for(:call)
+
+        def call(slug)
+          categories.delete(slug)
+        end
+
+        private
+
+        def prepare_attributes(attributes)
```
Comment (timriley) [inline apps/admin/lib/admin/categories/operations/delete.rb]: We don't need this in a delete operation?

File: apps/admin/lib/admin/views/categories/new.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    module Categories
+      class New < Admin::View
+        include Admin::Import("admin.categories.forms.create_form")
+
+        configure do |config|
+          config.template = "categories/new"
+        end
+
+        def locals(options = {})
+          super.merge(post_form: form_data(options[:validation]))
```
Comment (timriley) [inline apps/admin/lib/admin/views/categories/new.rb]: Probably good to bring this inline with our convention elsewhere and make it just `form:`?

File: apps/admin/lib/admin/views/categories/edit.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        end
+
+        def locals(options = {})
+          category = categories.by_slug(options[:slug])
+
+          category_validation = options[:category_validation]
+
+          super.merge(
+            category: category,
+            category_form: category_form(category, category_validation)
```
Comment (timriley) [inline apps/admin/lib/admin/views/categories/edit.rb]: Maybe we can just make this `form:` too?

File: apps/admin/lib/admin/views/categories/edit.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        )
+
+        configure do |config|
+          config.template = "categories/edit"
+        end
+
+        def locals(options = {})
+          category = categories.by_slug(options[:slug])
+
+          category_validation = options[:category_validation]
```
Comment (timriley) [inline apps/admin/lib/admin/views/categories/edit.rb]: and the same as our other conventions, make this just `:validation`?

**Comment by timriley** [comment]
Apart from the few minor things (and needing to use MethodOverride), this looks good!

**Pull request merged**